projects/sandbox/src/app/_services/upload-utils.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  AbstractControl,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  ValidationErrors,
+  Validators
+} from '@angular/forms';
+import { harvestValidator, ProtocolType } from 'shared';
+import { HarvestType } from '../_models';
+
+const formBuilder = new FormBuilder();
+
+export const harvestTypeToProtocolType = (harvestType: HarvestType): ProtocolType => {
+  if (harvestType === HarvestType.FILE) {
+    return ProtocolType.ZIP_UPLOAD;
+  } else if (harvestType === HarvestType.HTTP) {
+    return ProtocolType.HTTP_HARVEST;
+  } else {
+    return ProtocolType.OAIPMH_HARVEST;
+  }
+};
+
+/**
+ * validateDatasetName
+ *
+ * form validator implementation for dataset name field
+ *
+ * @param { FormControl } control - the control to validate
+ * @returns null or a code-keyed boolean
+ **/
+export const validateDatasetName = (control: FormControl<string>): ValidationErrors | null => {
+  const val = control.value;
+  if (val) {
+    const matches = /\w+/.exec(`${val}`); // NOSONAR
+    if (!matches || matches[0] !== val) {
+      return { invalid: true };
+    }
+  }
+  return null;
+};
+
+export const getNameSuggestion = (originalName: string): string => {
```

Comment:
Suggestion: Here is an opportunity to do a `random` dataset name generator to be filled in automatically by the app where users can turn on or off this feature so it saves filling the field.

add code documentation as the other methods.